perses/rjmc/topology_proposal.py

```diff
@@ -401,46 +423,82 @@ def _get_mol_atom_map(current_oemol,
         max_num_atoms = max([match.NumAtoms() for match in top_matches])
         _logger.debug(f"\tthere are {len(top_matches)} top matches with at most {max_num_atoms} before hydrogen exceptions")
         _logger.debug(f"\tthe max number of atom matches is: {max_num_atoms}; there are {len([m for m in top_matches if m.NumAtoms() == max_num_atoms])} matches herein")
-        
+
         # check the most mapped is the same once hydrogen exceptions are handled
         all_new_to_old_atom_maps = []
         count_after_hydrogen_mapping = []
-        for match in top_matches:
-            map_dict = AtomMapper.hydrogen_mapping_exceptions(current_oemol, proposed_oemol, match, matching_criterion) 
+        if return_all_maps:
+            list_of_dicts = []
+            for match in top_matches:
+                list_of_dicts.append(AtomMapper.hydrogen_mapping_exceptions(current_oemol, proposed_oemol, match, matching_criterion))
+            return list_of_dicts
+
+        for i, match in enumerate(top_matches):
+            map_dict = AtomMapper.hydrogen_mapping_exceptions(current_oemol, proposed_oemol, match, matching_criterion)
             count_after_hydrogen_mapping.append(len(map_dict))
             all_new_to_old_atom_maps.append(map_dict)
+            _logger.debug(f'index {i} = {map_dict}')
 
         max_num_atoms = max(count_after_hydrogen_mapping)
         _logger.info(f'Maximum atom matched after hydrogen exceptions: {max_num_atoms}')
-        new_to_old_atom_maps = [map for count, map in zip(count_after_hydrogen_mapping,all_new_to_old_atom_maps) if count == max_num_atoms]
-        
+        _logger.info(f'Number of maps is {len(all_new_to_old_atom_maps)}')
+
+        if geometry is not None:
```

Comment:
a bit nitpicky, but could this be a staticmethod that lives in the AtomMapper class?

Reply:
Done